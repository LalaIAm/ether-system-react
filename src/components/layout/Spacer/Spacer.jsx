import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import spacerSizes from '../../../base/spacing';

export const Spacer = ({ debug, horizontal, size, ...others }) => {
  const spacerBackground = debug ? 'pink' : 'none';
  const spacerPadding = horizontal ? 'padding-right' : 'padding-bottom';
  const spacerDisplay = horizontal ? 'inline-block' : 'block';
  const spacerHeight = (horizontal && debug) ? '5px' : '0';
  const StyledSpacer = styled.div`
    background-color: ${spacerBackground};
    display: ${spacerDisplay};
    height: ${spacerHeight};
    ${spacerPadding}: ${spacerSizes[size] || '0px'}
    vertical-align: middle;
  `;

  return (
    <StyledSpacer {...others} />
  );
};

export const SpacerSizes = ['none', '2xsmall', 'xsmall', 'small', 'medium', 'large', 'xlarge', '2xlarge'];

Spacer.propTypes = {
  /** If set, adds a light background color to the spacer for debugging purposes */
  debug: PropTypes.bool,
  /** Sets to horizontal spacing mode */
  horizontal: PropTypes.bool,
  /** Size of the spacer */
  size: PropTypes.oneOf(SpacerSizes),
};

Spacer.defaultProps = {
  size: 'medium',
};
