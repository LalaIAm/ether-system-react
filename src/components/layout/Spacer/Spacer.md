### Vertical Spacer
```js
<Spacer debug size="medium" />
```

### Horizontal Spacer
```js
const { Button } = require('../../input/Button/Button.jsx');
<span>
  <Button>Button 1</Button>
  <Spacer horizontal debug size="large" />
  <Button>Button 2</Button>
</span>
```
