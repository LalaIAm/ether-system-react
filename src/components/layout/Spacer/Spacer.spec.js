import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';

import { Spacer } from './Spacer';
import spacerSizes from '../../../base/spacing';

it('should match the snapshot', () => {
  const subject = shallow(<Spacer />);
  expect(subject).toMatchSnapshot();
});

it('should render inline-block if set to horizontal', () => {
  const subject = renderer.create(<Spacer horizontal />).toJSON();
  expect(subject).toMatchSnapshot();
  expect(subject).toHaveStyleRule('display', 'inline-block');
});

it('should receive correct padding based on size selection', () => {
  const subject = renderer.create(<Spacer size="large" />).toJSON();
  const mediumSpacerSize = spacerSizes.large;
  expect(subject).toMatchSnapshot();
  expect(subject).toHaveStyleRule('padding-bottom', mediumSpacerSize);
});
