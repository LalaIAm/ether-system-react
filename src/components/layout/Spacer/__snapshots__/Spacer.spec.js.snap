// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `<styled.div />`;

exports[`should receive correct padding based on size selection 1`] = `
.c0 {
  background-color: transparent;
  display: block;
  height: 0;
  padding-bottom: 36px;
  vertical-align: middle;
}

<div
  className="c0"
/>
`;

exports[`should render inline-block if set to horizontal 1`] = `
.c0 {
  background-color: transparent;
  display: inline-block;
  height: 0;
  padding-right: 23px;
  vertical-align: middle;
}

<div
  className="c0"
/>
`;
